### PR TITLE
chore: color docs for storybook

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

Add a docs page with the available global token colors.

Updated existing DocBlocks to use the styled-components tooling to mimic how native Storybook components are built ([example of the built-in ColorPalette component](https://github.com/storybookjs/storybook/blob/ae3b81c9f1f1f1aab4461ecca9345487e379ae2e/code/lib/blocks/src/components/ColorPalette.tsx)).

Screenshot 2024-08-09 at 9 58 29 AM

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

Open the [color palette](https://pr-2984--spectrum-css.netlify.app/preview/?path=/docs/guides-design-tokens--docs) page:

- [ ] Expect to see a documentation page with a gray container with rounded corners containing a set of color swatches aligning with our design token colors.
- [ ] Toggle the color theme in the top toolbar to dark and expect to see the gray box to update to a dark gray/black box with all the color swatches updated to their dark-mode variants.

Screenshot 2024-08-09 at 9 58 48 AM

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: .storybook/manager.js [hunk 1]
```diff
@@ -33,10 +36,9 @@ addons.setConfig({
 		textColor: "rgb(34, 34, 34)",
 
 		// Toolbar default and active colors
-		barTextColor:
-			"rgb(34, 34, 34)" /* --spectrum-neutral-content-color-default */,
-		barSelectedColor: "rgb(2, 101, 220)" /* --spectrum-global-color-blue-900 */,
-		// barBg: 'rgb(248, 248, 248)', /* --spectrum-alias-background-color-default */
+		barTextColor: "rgb(34, 34, 34)",
+		barSelectedColor: "rgb(2, 101, 220)",
+		// barBg: 'rgb(248, 248, 248)',
```
Comment (cdransf) [inline .storybook/manager.js:41]: ✨ Should we remove this if it's no being used?

**Review by cdransf — APPROVED** [review]

File: .storybook/manager.js (hunk 1 above)
Comment (castastrophe) [reply .storybook/manager.js:41]: I would but these values aren't well documented so I didn't want to forget that this keyword did something in the future.

**Review by jawinn — CHANGES_REQUESTED** [review]
Nice work on this! The new design tokens page with the swatches looks great, as do the custom components used for it.

I only noticed two things:
- I happened to inspect some of transparent white and black swatches that weren't visible and noticed that they're rendering an invalid property value. Do these not exist and need some handling for not setting the style? Probably related to the console warning "Token undefined can't be found in the spectrum token data":
Screenshot 2024-08-23 at 4 19 29 PM
- I checked the console and saw the ol' React "Warning: Each child in a list should have a unique "key" prop". Should be an easy fix.

File: .storybook/guides/color_palette.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+<br />
+
+### Transparent colors
+
+<ThemeContainer>
+  <ColorPalette>
+    <ColorItem
+      title="Transparent black"
+      color="transparent-black"
+      values={[25, 50, 75, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000]}
```
Comment (castastrophe) [inline .storybook/guides/color_palette.mdx:121]:
```suggestion
      values={[100, 200, 300, 400, 500, 600, 700, 800, 900]}
```

File: .storybook/guides/color_palette.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+</ThemeContainer>
+
+<br />
+
+<ThemeContainer>
+  <ColorPalette>
+    <ColorItem
+      title="Transparent white"
+      color="transparent-white"
+      values={[25, 50, 75, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000]}
```
Comment (castastrophe) [inline .storybook/guides/color_palette.mdx:133]:
```suggestion
      values={[100, 200, 300, 400, 500, 600, 700, 800, 900]}
```

**Comment by castastrophe** [comment]
> * I happened to inspect some of transparent white and black swatches that weren't visible and noticed that they're rendering an invalid property value. Do these not exist and need some handling for not setting the style?

Great catch!  I originally built this page for the S2 tokens to demonstrate the changes but I forgot to update the transparent-white and transparent-black ranges to the S1 set before opening the PR.  I've pushed up that fix.

File: .storybook/blocks/ColorPalette.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		<>
+			<SwatchGroupLabel className="swatch-group-label">
+				<Heading size="s">{title ?? capitalize(color)}</Heading>
+			</SwatchGroupLabel>
+			<SwatchGroup className="swatch-group" {...props}>
+				<SwatchColors className="swatch-colors" size={60}>
+					{values.map((value) => {
+						const resolved = fetchToken(`${color}-${value}`, value ?? color);
+						return (
+							<SwatchSet className="swatch-set">
```
Comment (jawinn) [inline .storybook/blocks/ColorPalette.jsx:97]:
```suggestion
							<SwatchSet className="swatch-set" key={`${color}-${value}`}>
```
That clears up the React key prop warning.

Comment (castastrophe) [reply .storybook/blocks/ColorPalette.jsx]: Nice!

**Review by jawinn — APPROVED** [review]

**Pull request merged**